lib/uri_service.rb

```diff
@@ -0,0 +1,15 @@
+module URIService
+  thread_mattr_accessor :active_connection
+
+  def self.connection
+    self.active_connection = new_connection unless active_connection
+    active_connection
+  end
+
+  def self.new_connection
+    config = Rails.application.config_for(:hyacinth)['uri_service']
```

Comment:
Rather than re-loading hyacinth.yml, you can re-use the `HYACINTH` constant that's made available in the `00_hyacinth.rb` initializer.